src/hooks/useSyncChainQuery.ts

```diff
@@ -31,8 +41,23 @@ export default function useSyncChainQuery() {
   const [searchParams, setSearchParams] = useSearchParams()
 
   useEffect(() => {
-    if (isActive && urlChainId && chainId !== urlChainId) {
+    // Change a user's chain on pageload if the connected chainId does not match the query param chain
+    if (isActive && urlChainId && chainIdRef.current === chainId && chainId !== urlChainId) {
       selectChain(urlChainId)
     }
-  }, [urlChainId, selectChain, searchParams, setSearchParams, isActive, chainId])
+    // If a user has a connected wallet and has manually changed their chain, delete the old chain query param
+    else if (account && chainIdRef.current !== chainId && chainId !== urlChainId) {
+      if (isSupportedChain(chainId)) {
+        searchParams.set('chain', CHAIN_IDS_TO_NAMES[chainId])
+      } else {
+        searchParams.delete('chain')
+      }
```

Comment:
This logic does not match the comment:

```suggestion
      searchParams.delete('chain')
```

Reply:
Comment should be updated in stead, thanks for the callout